core/src/main/java/com/klaytn/caver/tx/type/TxTypeFeeDelegate.java

```diff
@@ -16,9 +16,160 @@
 
 package com.klaytn.caver.tx.type;
 
+import com.klaytn.caver.crypto.KlayCredentials;
+import com.klaytn.caver.crypto.KlaySignatureData;
+import com.klaytn.caver.tx.exception.EmptyNonceException;
+import com.klaytn.caver.tx.model.KlayRawTransaction;
+import com.klaytn.caver.utils.BytesUtils;
+import org.web3j.rlp.RlpEncoder;
+import org.web3j.rlp.RlpList;
+import org.web3j.rlp.RlpString;
+import org.web3j.rlp.RlpType;
+import org.web3j.utils.Numeric;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
 /**
  * This interface represents which is fee delegated transaction type
  */
-public interface TxTypeFeeDelegate {
+public abstract class TxTypeFeeDelegate extends AbstractTxType {
+    final static String EMPTY_FEE_PAYER_ADDRESS = "0x30";
+
+    private Set<KlaySignatureData> feePayerSignatureData;
+    private String feePayer;
+
+    public TxTypeFeeDelegate(BigInteger nonce, BigInteger gasPrice, BigInteger gasLimit,
+                             String from, String to, BigInteger value) {
+        super(nonce, gasPrice, gasLimit, from, to, value);
+        this.feePayerSignatureData = new HashSet<>();
+        this.feePayer = EMPTY_FEE_PAYER_ADDRESS;
+    }
+
+    public Set<KlaySignatureData> getFeePayerSignatureData() {
+        return feePayerSignatureData;
+    }
+
+    public String getFeePayer() {
+        return this.feePayer;
+    }
+
+    public void setFeePayer(String feePayer) {
+        this.feePayer = feePayer;
+    }
+    
+    public BigInteger getFeeRatio() {
+        return BigInteger.valueOf(100);
```

Comment:
Where does txType use this method? I think txType end with 'WithRatio' has this method.

Reply:
If there is no ratio, the fee payer pays 100 percent.

Reply:
The TxTypeFeeDelegate type allows all sub class to use a common interface with or without ratio. So it's why this method made.